include/caf/events/module_state_event.h

```diff
@@ -60,6 +60,25 @@ static inline const void *module_id_get(size_t idx)
 	return *((&MODULE_ID_LIST_START) + idx);
 }
 
+/**
+ * @brief Get IDX of module with given id.
+ *
+ * @param[in] id ID of the module.
+ *
+ * @return Module IDX.
+ */
+static inline size_t module_idx_get(const void *id)
+{
+	for (size_t idx = 0; idx < module_count(); idx++) {
```

Comment:
It would be great to get rid of this for loop if possible (to speed up conversion from id to idx). Discussed offline.